### PR TITLE
Preliminary Hamburger menu, login screen

### PR DESCRIPTION
**Pull request by ArchKevin22**
Yeah, it's ugly I know. I'll work on the styling of it later.

### CODE REVIEW

**Comment by ArchKevin22** [comment]
I don't know if package-lock.json will cause a shitload of merge conflicts for all of you, but it did for me.

File: client/js/HomeScreen.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    };
+    constructor(props) {
+        super(props);
+        this.state = {
+            user_id: '',
+            password: '',
+            session_key: ''
+        };
+
+        this.closeDrawer = this.closeDrawer.bind(this);
```
Comment (bliu23) [inline client/js/HomeScreen.js]: do you have to bind `this` for all functions? seems redundant to bind if you dont need to

File: client/js/HomeScreen.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    provider() {
+        //load nearby requests screen
+    }
+    requester() {
+        this.props.navigation.navigate('MakeRequest')
+    }
+
+    _setNavigationParams() {
+        let headerLeft  = <Button
+        //style={styles.button}
```
Comment (bliu23) [inline client/js/HomeScreen.js]: Button styles are really whack so dw about this for now

File: client/js/HomeScreen.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        title='Menu'
```
Comment (bliu23) [inline client/js/HomeScreen.js]: nit: indentation

File: client/js/HomeScreen.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+          headerLeft,
+        });
+    }
+
+    componentWillMount() {
+        this._setNavigationParams();
+      }
+
+    render() {
+        const drawerStyles = {
```
Comment (bliu23) [inline client/js/HomeScreen.js:82]: we'll need to move the styles somewhere better eventually (all styles in general)

File: client/js/HomeScreen.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                      }}/>}
+                    ref={(ref) => this._drawer = ref}
+                    openDrawerOffset={0.6}
+                    style={drawerStyles}
+                    tapToClose={true}
+                    acceptPan={true}
+                    panCloseMask={0.9}
+                    panOpenMask={0}
+                >   
+                    <View
```
Comment (bliu23) [inline client/js/HomeScreen.js]: nit: you can prob put this on the same line for readability

**Review by bliu23 — APPROVED** [review]
l o o k s gr8

**Pull request merged**